blaze/compute/sql.py

```diff
@@ -585,6 +593,24 @@ def compute_up(expr, data, **kwargs):
     return table_of_engine(data, expr._name)
 
 
+@dispatch(DateTime, (ClauseElement, sa.sql.elements.ColumnElement))
+def compute_up(expr, data, **kwargs):
+    return sa.extract(expr.attr, data).label(expr._name)
+
+
+@compiles(sa.sql.elements.Extract, 'hive')
```

Comment:
the `compiles` decorator lets you selectively compile things differently based on the dialect and the expression type